R/workbook_write_data.R

```diff
@@ -227,6 +227,33 @@ Workbook$methods(writeData = function(df, sheet, startRow, startCol, colNames, c
   newStrs <- v[strFlag]
   if (length(newStrs) > 0) {
     newStrs <- replaceIllegalCharacters(newStrs)
+    vl <- stri_length(newStrs)
+    
+    for (i in which(vl > 32767)) {
+      
+      if(vl[i]>32768+30){
+        warning(
+          paste0(
+            stri_sub(newStrs[i], 32768, 32768 + 15),
+            " ... " ,
+            stri_sub(newStrs[i], vl[i] - 15, vl[i]),
+            " is truncated. 
+Number of characters exeed the limit of 32767."
+          )
+        )
+      } else {
+        warning(
+          paste0(
+            stri_sub(newStrs[i], 32768, -1),
+            " is truncated. 
+Number of characters exeed the limit of 32767."
+          )
+        )
+        
+      }
+      
+      # v[i] <- stri_sub(v[i], 1, 32767)
```

Comment:
isn't something like this required?

Reply:
I had the following in mind:
- for data which is just a bit over the limit it is shows the truncated part
- for data which is way over the limit the warning shows the head and the tail of the truncated data
- this additional information doesn't have a big impact on the performance, but I would like to know which infromation I might not have in my workbook.